src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -173,9 +175,9 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 		this._updateUnicodeVersion();
 		this._commandNavigationAddon = this._instantiationService.createInstance(CommandNavigationAddon, _capabilities);
 		this.raw.loadAddon(this._commandNavigationAddon);
-		this._shellIntegrationAddon = this._instantiationService.createInstance(ShellIntegrationAddon);
+		this._shellIntegrationAddon = new ShellIntegrationAddon(this._telemetryService, this._logService);
```

Comment:
And here we can avoid touching log service, leave it up to instantiation service.

```suggestion
		this._shellIntegrationAddon = this._instantiationService.createInstance(ShellIntegrationAddon, this._telemetryService);
```

Not a big deal though, can push without it